SpatialGDK/Extras/git/common/pre-commit

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+# Hit em with that clang-format
+for FILE in $(git diff --name-only)
+do
+    clang-format -verbose -i $FILE
```

Comment:
```suggestion
    clang-format -verbose -i "${FILE}"
```

Reply:
and this should probably check if you have `clang-format` installed. it's not  installed by default right?

Reply:
just checking: on mac the verbose flag is: `--verbose` . is it `-verbose` on windows?

Reply:
Ye it's `-verbose` - do we care about macos though? (just kidding will change)

Reply:
Ye, should clarify I'm yet to explore installation options, just wanting us to get to a point where we agree having it is good and agreeable 😄

Reply:
> just checking: on mac the verbose flag is: --verbose . is it -verbose on windows?

Turns out both work on windows to will just do double-dash

Reply:
Added a TODO into PR description so it's clear am still figuring out how to get clang-format on user machines 🙂